flink-formats/flink-orc/src/test/java/org/apache/flink/orc/writer/OrcBulkWriterTest.java

```diff
@@ -75,8 +78,105 @@ public void testOrcBulkWriter() throws Exception {
 
             testHarness.snapshot(1, ++time);
             testHarness.notifyOfCompletedCheckpoint(1);
+        }
+    }
 
-            OrcBulkWriterTestUtil.validate(outDir, input);
+    @Test
+    public void testOrcBulkWriter() throws Exception {
+        final File outDir = TEMPORARY_FOLDER.newFolder();
+        final Properties writerProps = new Properties();
+        writerProps.setProperty("orc.compress", CompressionKind.LZ4.name());
+
+        OrcBulkWriterFactory<Record> writer =
+                new OrcBulkWriterFactory<>(
+                        new RecordVectorizer(schema), writerProps, new Configuration());
+
+        writeRecordsIntoOrcFile(outDir, writer);
+
+        // validate records and compression kind
+        OrcBulkWriterTestUtil.validate(outDir, input, CompressionKind.LZ4);
+    }
+
+    @Test
+    public void testZstdCompression() throws Exception {
+        final File outDir = TEMPORARY_FOLDER.newFolder();
+        final Properties writerProps = new Properties();
+        writerProps.setProperty("orc.compress", CompressionKind.ZSTD.name());
+
+        OrcBulkWriterFactory<Record> writer =
+                new OrcBulkWriterFactory<>(
+                        new RecordVectorizer(schema), writerProps, new Configuration());
+
+        writeRecordsIntoOrcFile(outDir, writer);
+
+        // validate records and compression kind
+        OrcBulkWriterTestUtil.validate(outDir, input, CompressionKind.ZSTD);
+    }
+
+    @Test
+    public void testOrcColumnEncryption() throws Exception {
+        final File outDir = TEMPORARY_FOLDER.newFolder();
+
+        // Simple configuration for column encryption.
+        Configuration conf = new Configuration();
+        conf.set("orc.encrypt", "pii:_col0");
+        conf.set("orc.mask", "sha256:_col0");
+        OrcFile.WriterOptions writerOptions =
+                OrcFile.writerOptions(conf)
+                        .encrypt(OrcConf.ENCRYPTION.getString(conf))
+                        .masks(OrcConf.DATA_MASK.getString(conf))
+                        .setKeyProvider(new DummyInMemoryKeystore());
+
+        OrcBulkWriterFactory<Record> writer =
+                new OrcBulkWriterFactory<>(new RecordVectorizer(schema), writerOptions);
+
+        writeRecordsIntoOrcFile(outDir, writer);
+
+        // Validate bucket count and file count.
```

Comment:
better to extract this part into a method, instead of the comment, see the structure of other tests here

Reply:
done